### DISCUSSION TITLE
请教下计算PPL的方法：

### DISCUSSION THREAD

**Discussion by niexufei** [open-compass/opencompass#140] [Q&A] [answered]
opencompass-3715be6/opencompass/models/huggingface.py中的

`def _get_ppl(self, inputs: List[str], mask_length: Optional[List[int]] = None) -> List[float]:    
    outputs, inputs = self.get_logits(inputs)
    .......
`
这里给你一个input列表，就可以计算出PPL的值；
首先调用get_logits，返回模型预测的token和input token
这个input 就是按照模板组成的prompt，以ceval为例，就是five-shot的例子加上新的题目以及答案是一个label “A”
我的问题是：为啥将这个input给模型，调用模型的前向传播方法，获取output[0]就是模型的预测呢？
“fiveshot + 新的选择题目+answer ：A“  这个给模型是想让模型预测啥呢？

大模型小白，希望高手赐教。

**Comment by yingfhu** [comment, marked as answer]
refers to https://github.com/InternLM/opencompass/issues/150 for more discussions.
